File: easy/0389_find_the_difference/solution.cpp
#include <string>
#include <unordered_map>

class Solution {
public:
  char findTheDifference(std::string s, std::string t) {
    char diff = 0;

    for (const char &c : s)
      diff ^= c;

    for (const char &c : t)
      diff ^= c;

    return diff;
  }
};

File: easy/0389_find_the_difference/solution.py
from collections import Counter


class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        s_counter = Counter(s)

        for c in t:
            if c in s_counter:
                s_counter[c] -= 1

                if s_counter[c] < 0:
                    return c
            else:
                return c
